components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/validator/DefaultApplicationValidator.java

```diff
@@ -305,6 +320,49 @@ private void validateRequestPathAuthenticationConfig(List<String> validationMsg,
         }
     }
 
+    /**
+     * Validate trusted app related configurations and append to the validation msg list.
+     *
+     * @param validationMsg   validation error messages.
+     * @param serviceProvider service provider.
+     */
+    private void validateTrustedAppMetadata(List<String> validationMsg, ServiceProvider serviceProvider) {
+
+        SpTrustedAppMetadata trustedAppMetadata = serviceProvider.getTrustedAppMetadata();
+        if (trustedAppMetadata == null) {
+            return;
+        }
+
+        // Validate if feature is enabled without data.
+        if (trustedAppMetadata.getIsFidoTrusted() && StringUtils.isBlank(trustedAppMetadata.getAndroidPackageName()) &&
+                StringUtils.isBlank(trustedAppMetadata.getAppleAppId())) {
+            validationMsg.add(TRUSTED_APP_FEATURE_ENABLED_WITHOUT_DATA);
+        }
+
+        // Validate the android thumbprints count.
+        if (StringUtils.isNotBlank((trustedAppMetadata.getAndroidThumbprints()))) {
+            String[] thumbprints = trustedAppMetadata.getAndroidThumbprints().split(ATTRIBUTE_SEPARATOR);
+            if (thumbprints.length > MAX_ANDROID_THUMBPRINT_COUNT) {
+                validationMsg.add(MAX_THUMBPRINT_COUNT_EXCEEDED);
+            }
+        }
+
+        // Validate consent for trusted apps.
+        if (Boolean.parseBoolean(IdentityUtil.getProperty(TRUSTED_APP_CONSENT_REQUIRED_PROPERTY)) &&
+                trustedAppMetadata.getIsFidoTrusted() &&
+                !ApplicationMgtUtil.isTrustedAppConsentGranted(serviceProvider)) {
+            validationMsg.add(TRUSTED_APP_NOT_CONSENTED);
+        }
+
+        // Validate the android app details.
+        if ((StringUtils.isNotBlank(trustedAppMetadata.getAndroidPackageName()) &&
+                StringUtils.isBlank(trustedAppMetadata.getAndroidThumbprints())) ||
+                (StringUtils.isBlank(trustedAppMetadata.getAndroidPackageName()) &&
+                        StringUtils.isNotBlank(trustedAppMetadata.getAndroidThumbprints()))) {
+            validationMsg.add(INCORRECT_TRUSTED_ANDROID_APP_DETAILS);
+        }
+    }
+
```

Comment:
Add a validation to fail if consent is set to false without disabling the feature

Reply:
It is already covered by the previous validation 3 and the error message is updated as:
"Consent should be granted for trusted apps if FIDO trusted app feature is enabled."